File: ios/CustomControls/CustomControls/Glasses/CustomControlsScreen.swift
//Copyright (c) 2020 Everysight LTD

import Foundation
import NativeEvsKit


class CustomControlsScreen : Screen {
    
    private var text = Text()
   
    override func onCreate() {
        add(uiElement: SimpleControl().setX(x: 30).setY(y: 30))
        add(uiElement: SimpleControl().setX(x: 30).setY(y: 100))

        text
            .setText(text: "Custom Controls")
            .setResource(font: Font(stockFont: .medium))
            .setTextAlign(align: Align.center)
        text
            .setX(x: getWidth()/2).setY(y: getHeight()-40)
            .setForegroundColor(color: EvsColor.green.rgba)
        add(uiElement: text)

        let clock = AnalogClockControl()
        clock
            .setX(x: getWidth()/2-50.0)
            .setY(y: 30.0)
            .setWidth(width: 150.0)
            .setHeight(height: 150.0)
        add(uiElement: clock)
        
    }
}

File: android/CustomControls/src/main/java/com/everysight/sdk/samples/customcontrols/CustomControlsScreen.kt
package com.everysight.sdk.samples.customcontrols

import UIKit.app.Screen
import UIKit.app.data.Align
import UIKit.app.data.EvsColor
import UIKit.app.resources.Font
import UIKit.widgets.Text

class CustomControlsScreen:Screen() {

    private val text = Text()
    companion object{
        const val SZ_PIX = 150f
    }

    override fun onCreate() {
        add(SimpleControl().setX(30f).setY(30f))
        add(SimpleControl().setX(30f).setY(100f))

        text.setText("Custom Controls").setResource(Font.StockFont.Medium).setTextAlign(Align.center)
        text.setX(getWidth()/2).setY(getHeight()-40).setForegroundColor(EvsColor.Green.rgba)
        add(text)

        val clock = AnalogClockControl()
        clock.setX((getWidth()- SZ_PIX)/2).setY((getHeight()-SZ_PIX)/2).setWidth(SZ_PIX).setHeight(SZ_PIX)
        add(clock)
    }

}